### ISSUE TITLE
Please clarify: Why remove connection from group?

### ISSUE THREAD

**Issue by HappyNomad** [aspnet/Docs#10617]
The `OnDisconnectedAsync` example in the [Handle events for a connection
](https://docs.microsoft.com/en-us/aspnet/core/signalr/hubs?view=aspnetcore-2.2#handle-events-for-a-connection) section removes the connection from the group.  But in the "Hubs" doc's [Groups in SignalR](https://docs.microsoft.com/en-us/aspnet/core/signalr/groups?view=aspnetcore-2.2#groups-in-signalr) section it says, "Group membership isn't preserved when a connection reconnects. The connection needs to rejoin the group when it's re-established."

As the reader of these docs, trying to make sense of them, I'm left wondering: Why remove a connection from a group when it'll be removed anyway?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1387f556-b4b7-70f9-5bd0-f0bccc597689
* Version Independent ID: fb016cdc-d281-6b5b-7466-2c718e7274b4
* Content: [Use hubs in ASP.NET Core SignalR](https://docs.microsoft.com/en-us/aspnet/core/signalr/hubs?view=aspnetcore-2.2#handle-events-for-a-connection)
* Content Source: [aspnetcore/signalr/hubs.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/signalr/hubs.md)
* Product: **aspnet-core**
* GitHub Login: @bradygaster
* Microsoft Alias: **bradyg**

**Comment by bradygaster** [comment]
@HappyNomad each time a connection is made, it gets an individual connection id specific to that connection's lifetime. When we speak of a "connection" it is less than a user - an individual user might connect to a hub N times during an individual session - wifi going out, timeout on the server, a variety of reasons that can't be controlled (like a mobile phone going out of signal range and then coming back into it, thus the client would reconnect if the developer had added reconnection logic). 

On the other hand, clients might stay connected for considerable lengths of time. In that case, there might be a legitimate reason why your logic would **want** to remove a connection from a group. Let's say a connection is in a chat room and they leave the room to go to another one. In that case, the "room" would be the "group," and it would be desirable to have the functionality to add/remove connections from a group. 

Does that help make sense of why this type of code/guidance exist?

**Comment by HappyNomad** [comment]
@bradygaster Thanks but clarification is still eluding me.

I understand your "chat room" use case, and that it's "a legitimate reason why your logic would want to remove a connection from a group".  What about disconnection?  The docs have an `OnDisconnectedAsync` example rather than a chat room one.  Is removing a connection from a group upon disconnection "legitimate", necessary, or useful?  Or is it instead "illegitimate", unnecessary, or useless?

I'm asking because I also need to add/remove a connection to/from a group upon connection/disconnection.  Adding is easy but the conflicting information in the docs is confusing me: Do I need to manually remove it upon disconnection or not?

**Comment by bradygaster** [comment]
TL/DR version: Removing a connection from a group during the disconnected handler is a little unnecessary given that connection, once removed, won't be in the group any longer. 

Apologies if the docs appear conflicting, that isn't intentional. Both of the above articles demonstrate adding and removing connections from groups. That's the main reason - to demonstrate adding and removing. I agree that showing the removal-from-group part during the method that "does it automatically anyway" isn't as compelling. I **think** there might be a better line of code from the existing sample that could be referenced there. I'll look into it. 

Thanks for the feedback, you make a good point - that the call at the place it is being made might be moot.

**Comment by bradygaster** [comment]
Closing this as we haven't heard from you. Please feel free to comment if you're able to get the information we're looking for and we can reopen the issue if we're able to identify the problem.

**Comment by ChainReactive** [comment]
@bradygaster Your reply last week answered my question.  Perhaps you also want to clarify it in the docs?  You made a good case for doing so in your own comment, so I guess it's just a matter of you following through or not.  In any case, thanks for your help.

**Comment by bradygaster** [comment]
Good point, and we had a discussion about it during stand-up. The place where we show you how to leave a group isn't the best place. :)

**Comment by Tommigun1980** [comment]
If you want to show how removing of connections work having it in the disconnect method is probably the most confusing place imaginable, and makes it look like it’s a mandatory step, especially when the connection is added to a group in the connect method. Thank you @bradycaster for the heads up.
